src/range_query/sqrt_decomp.rs

```diff
@@ -129,14 +131,54 @@ impl PiecewiseLinearFn {
     fn update_envelope(&mut self) {
         self.recent_lines.extend(self.sorted_lines.drain(..));
         self.recent_lines.sort_unstable();
```

Comment:
If we're using floating arithmetic, we might as well change everything to `f64`. I suppose that would mean changing this line to something like
`self.recent_lines.sort_unstable_by(|a, b| a.partial_cmp(b).unwrap());`

Reply:
Intuitively I think of the lines as something more user facing so it makes sense to keep them in the original form, whereas the intersection points are more of an implementation detail so we can make those floats, or whatever helps us implement the algorithm. Rational numbers could work but they're probably less efficient. I guess most of the time our line coefficients will fit into an int32 so f64 wouldn't lose any precision.

Reply:
I'm somewhat partial to keeping it consistent: since we're not promising to deliver exact arithmetic, `f64` seems like the more general interface. I think I'll update lower/upper_bound and add a comparator, to make dealing with floats less annoying in cases where the user is willing to tolerate a panic when elements fail to be comparable. I'll hold off on updating my branch while this PR is active. If you feel like including them in your branch, these are the changes:

lower/upper_bound will take `T: PartialOrd` and call `.partial_cmp().unwrap()` instead of `cmp()`.
And here's the comparator:
```
fn asserting_cmp<T: PartialOrd>(a: &T, b: &T) -> std::cmp::ordering { 
    a.partial_cmp(b).unwrap()
}
```
With this, I think you can sort using `self.recent_lines.sort_unstable_by(asserting_cmp);`